### DISCUSSION TITLE
How to get Mastodon’s user ID from Mastodon account?

### DISCUSSION THREAD

**Discussion by patrickmcurry** [mastodon/mastodon#21156] [General]
Per this thread on the Mastodon Meta Discussion Board...
https://discourse.joinmastodon.org/t/how-to-get-mastodons-user-id-from-mastodon-account-acct-username-server/1658

I can't post a new reply there, but in lieu of the v2 API call to `/api/v2/search` I was able to do this with a call to `/api/v1/accounts/lookup?acct=[username]`

Any downsides to doing it this way?

**Comment by trwnh** [comment]
I think accounts/lookup doesn't resolve remote accounts, so it only works if the account is already known

**Reply by patrickmcurry** [reply]
Ok thanks for the info!
